### ISSUE TITLE
Bug: Link without any other text fails to compact

### ISSUE THREAD

**Issue by nelsonic** [dwyl/link#8]
As noted by @LuchoTurtle in https://github.com/dwyl/link/pull/5#pullrequestreview-1563500851 there is currently an issue with whitespace in the demo/showcase of this package: https://amemo.fly.dev/

![image](https://github.com/dwyl/link/assets/194400/fc869eea-5c6b-4170-a551-720ff6bb868f)

Try it:
```md
https://github.com/dwyl/link/pull/5#pullrequestreview-1558913764 
```

# Todo

+ [x] Create a test that has only a URL and no other text or whitespace.
+ [ ] Make the test pass!

**Comment by nelsonic** [comment]
@LuchoTurtle I've added a test and _two_ doctests. Both _pass_ without modifying any code. 💭 
Perhaps you can help with defining a test for the condition you have identified in your screen recording. 🙏

**Comment by nelsonic** [comment]
https://github.com/dwyl/link/blob/d342f8fc83a67077b483f10bee906894e2578c28/lib/link.ex#L207-L208

**Comment by nelsonic** [comment]
Working on `localhost`
![image](https://github.com/dwyl/link/assets/194400/c4638bf2-dbfb-4da4-83cf-1e014f1ad912)

**Comment by nelsonic** [comment]
@LuchoTurtle PR links Working flawlessly on: https://amemo.fly.dev/
![pr-links-still-work](https://github.com/dwyl/link/assets/194400/59a638cb-48c6-48fb-b2d8-eded927a74d6)

This is included in PR #5

**Comment by LuchoTurtle** [comment]
Confirmed 👌 . 

<img width="966" alt="image" src="https://github.com/dwyl/link/assets/17494745/4ee905fd-8f8e-484d-af51-a07223aeec79">

Thanks!
